src/main/java/Duke.java

```diff
@@ -1,10 +1,133 @@
+
+        // while loop to repeat printing of multiple Scanner inputs
+        // adapted with help of AI
+        while (true) {
+            String input = sc.nextLine();
+
+            try {
+                // Check if the user wants to exit.
+                // When comparing strings for equality, you should use the equals() method, not the == operator.
+                if (input.equals("bye")) { // exit chat
+                    System.out.println("Bye. It's been a pleasure chatting with you!");
+                    break;
+                } else if (input.equals("list")) { // list tasks
+                    System.out.println("Here are the tasks in your list:");
+
+                    for (int i = 0; i < tasks.size(); i++) {
+                        Task currTask = tasks.get(i);
+                        System.out.println((i + 1) + ". " + currTask);
+                    }
+                } else if (input.startsWith("mark")) {
+                    Integer indexOfTask = Integer.parseInt(input.substring(5));
+                    Task currTask = tasks.get(indexOfTask);
+                    currTask.markAsDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(currTask);
+                } else if (input.startsWith("unmark")) {
+                    Integer indexOfTask = Integer.parseInt(input.substring(7));
+                    Task currTask = tasks.get(indexOfTask);
+                    currTask.markAsNotDone();
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println(currTask);
+                } else if (input.startsWith("todo")) {
+                    // Check if the input string is long enough
+                    if (input.length() <= 5) {
+                        throw new DukeException("Please specify the description of the todo task, " +
+                                "starting from one whitespace away from the keyword 'todo'" +
+                                " (e.g. todo borrow book)");
+                    }
+
+                    String description = input.substring(5);
+
+                    if (description.isEmpty()) {
+                        throw new DukeException("Please specify the description of the todo task, " +
+                                "starting from one whitespace away from the keyword 'todo'" +
+                                " (e.g. todo borrow book)");
+                    }
+
+                    Task todo = new Todo(description);
+                    tasks.add(todo);
+                    System.out.println("Got it. I've added this task: \n" + todo);
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } else if (input.startsWith("deadline")) {
+                    String[] substrings = input.split(" /by ");
+
+                    if (substrings[0].length() <= 9) {
```

Comment:
Really good break (empty space) between each logical unit. It makes it so easy to understand your code!